src/unity/lib/visualization/plot/plot.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <memory>
+
+#include "plot.hpp"
+#include <unity/lib/visualization/process_wrapper.hpp>
+#include <unity/lib/visualization/thread.hpp>
+#include <unity/lib/visualization/transformation.hpp>
+#include <unity/lib/visualization/vega_data.hpp>
+#include <unity/lib/visualization/histogram.hpp>
+#include <unity/lib/visualization/item_frequency.hpp>
+#include <unity/lib/visualization/summary_view.hpp>
+
+namespace turi{
+  namespace visualization{
+    void Plot::show() {
+      using namespace turi;
+      using namespace turi::visualization;
+
+      std::shared_ptr<Plot> self = std::make_shared<Plot>(*this);
+
+      ::turi::visualization::run_thread([self]() {
+        process_wrapper ew(self->path_to_client);
+        ew << self->vega_spec;
+
+        while(ew.good()) {
+          vega_data vd;
+
+          vd << self->transformer.get()->get()->vega_column_data();
+
+          double num_rows_processed =  static_cast<double>(self->transformer->get_rows_processed());
+          double percent_complete = num_rows_processed/self->size_array;
+
+          ew << vd.get_data_spec(percent_complete);
+
+          if (self->transformer->eof()) {
+             break;
+          }
+        }
+      });
+    }
+
+    void Plot::materialize() {
+      do {
+        transformer.get()->get()->vega_column_data();
```

Comment:
You should be able to just do `transformer->get()`. The first get is to resolve the shared pointer to a raw pointer (which arrow syntax does automatically), and the vega_column_data is unnecessary here, I think.

Reply:
changed